astropy/coordinates/tests/test_geodetic_representations.py

```diff
@@ -2,49 +2,116 @@
 
 """Test geodetic representations"""
 import pytest
-from numpy.testing import assert_array_equal
 
 from astropy import units as u
-from astropy.coordinates.earth import (
+from astropy.coordinates.geodetic import (
+    BaseGeodeticRepresentation,
     GRS80GeodeticRepresentation,
     WGS72GeodeticRepresentation,
     WGS84GeodeticRepresentation,
 )
-from astropy.coordinates.representation import CartesianRepresentation
+
+from astropy.coordinates.representation import (
+    CartesianRepresentation,
+    REPRESENTATION_CLASSES,
+)
 from astropy.units import allclose as quantity_allclose
+from astropy.units.tests.test_quantity_erfa_ufuncs import vvd
+
+# Preserve the original REPRESENTATION_CLASSES dict so that importing
+#   the test file doesn't add a persistent test subclass (LogDRepresentation)
+from astropy.coordinates.tests.test_representation import (
+    setup_function,
+    teardown_function,
+)
+
+
+def test_custombasegeodetic_error():
+    # Test incomplete initialization.
+
+    msg = "module 'erfa' has no attribute 'foo'"
+    with pytest.raises(AttributeError, match=msg):
 
+        class CustomGeodeticEllipsoidError(BaseGeodeticRepresentation):
+            _ellipsoid = "foo"
 
-def test_cartesian_wgs84geodetic_roundtrip():
+    assert "customgeodeticellipsoiderror" not in REPRESENTATION_CLASSES
+
+    msg = "CustomGeodeticError requires '_ellipsoid' or '_equatorial_radius' and '_flattening'."
+    with pytest.raises(AttributeError, match=msg):
+
+        class CustomGeodeticError(BaseGeodeticRepresentation):
+            _flattening = 0.075 * u.dimensionless_unscaled
+
+    assert "customgeodeticerror" not in REPRESENTATION_CLASSES
+
+
+def test_custombasegeodetic_roundtrip():
```

Comment:
This test should be [parametrized](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions). If one of the asserts fails then the following asserts in the test function are not executed, but in the case of a parametrized test one of the parametrizations failing does not prevent other parametrizations of the same test from running, so a parametrized test suite is more granular and informative with no or very little extra effort.